### ISSUE TITLE
Add support for notifying into matrix

### ISSUE THREAD

**Issue by telmich** [fluxcd/notification-controller#229]
Matrix (matrix.org) is an open source federated chat is more and more used as a replacement for Slack.
Matrix has a very simple API that is alread implemented by services such as gitea.

The typical parameters for a matrix webhook are:

* Homeserver URL
* Room ID (where to post)
* Access token (authentication)
* Event type (optional, can be m.notice or m.text)

**Comment by stefanprodan** [comment]
You could use the Slack provider along with https://matrix.org/docs/projects/as/matrix-appservice-webhooks

**Comment by telmich** [comment]
That is absolutely correct, @stefanprodan. However that would mean everyone using Matrix will need to add their own webhook server. That's why I was wondering if it makes sense to add this functionality directly into flux.

**Comment by stefanprodan** [comment]
@telmich I'm not against implementing an alert provider for Matrix, I just wanted to add a note here for others, that there is a workaround for using Matrix even if Flux don't supports yet.
